src/Settings.pm

```diff
@@ -428,14 +428,26 @@ sub loadByHandle {
 	assert(defined $object) if DEBUG;
 
 	my $filename;
-	if ($object->{autoSearch}) {
-		if ($object->{type} == CONTROL_FILE_TYPE) {
-			$filename = _findFileFromFolders($object->{name}, \@controlFolders);
+	my $internalFilename;
+	if ($object->{'autoSearch'}) {
+		$internalFilename = $object->{'name'};
+	} else {
+		$internalFilename = $object->{'internalName'};
+	}
+
+	#hooks of type 'pre_load_' make it possible to change the filename before openkore searches for it (this filename must contain file name and file path)
+	my %pre_load = (internalFilename => $internalFilename, filename => \$filename);
+	Plugins::callHook('pre_load_'.$internalFilename, \%pre_load);
```

Comment:
$filename = $pre_load->{filename}; is not necessary since in my code $pre_load{'filename'} holds a reference to '$filename' so it can be changed inside the plugin itself without any extra source code. About the nested logic, i guess since we are at it there could be no bad at doing some cleaning.